diff --git a/node_modules/@vitejs/plugin-react/dist/index.mjs b/node_modules/@vitejs/plugin-react/dist/index.mjs
index 3022f48..ded0e99 100644
--- a/node_modules/@vitejs/plugin-react/dist/index.mjs
+++ b/node_modules/@vitejs/plugin-react/dist/index.mjs
@@ -244,7 +244,7 @@ function viteReact(opts = {}) {
       const [filepath, querystring = ""] = id.split("?");
       const [extension = ""] = querystring.match(fileExtensionRE) || filepath.match(fileExtensionRE) || [];
       if (/\.(mjs|[tj]sx?)$/.test(extension)) {
-        const isJSX = extension.endsWith("x");
+        const isJSX = extension.endsWith("x") || code.includes('React');
         const isNodeModules = id.includes("/node_modules/");
         const isProjectFile = !isNodeModules && (id[0] === "\0" || id.startsWith(projectRoot + "/"));
         let babelOptions = staticBabelOptions;
