diff --git a/node_modules/vite/dist/node/chunks/dep-934dbc7c.js b/node_modules/vite/dist/node/chunks/dep-934dbc7c.js
index d3d3ed5..34f6447 100644
--- a/node_modules/vite/dist/node/chunks/dep-934dbc7c.js
+++ b/node_modules/vite/dist/node/chunks/dep-934dbc7c.js
@@ -40942,7 +40942,7 @@ function transformCjsImport(importExp, url, rawUrl, importIndex, importer, confi
                 lines.push(`const ${localName} = ${cjsModuleName}`);
             }
             else if (importedName === 'default') {
-                lines.push(`const ${localName} = ${cjsModuleName}.__esModule ? ${cjsModuleName}.default : ${cjsModuleName}`);
+                lines.push(`const ${localName} = ${cjsModuleName}.__esModule && ${cjsModuleName}.default ? ${cjsModuleName}.default : ${cjsModuleName}`);
             }
             else {
                 lines.push(`const ${localName} = ${cjsModuleName}["${importedName}"]`);
